core/common/src/main/java/alluxio/security/authorization/DefaultAccessControlList.java

```diff
@@ -60,27 +59,46 @@ public DefaultAccessControlList(AccessControlList acl) {
 
   /**
    * create a child file 's accessACL based on the default ACL.
+   * @param mode file's initial mode based on umask
    * @return child file's access ACL
    */
-  public AccessControlList generateChildFileACL() {
+  public AccessControlList generateChildFileACL(Short mode) {
+    Mode defaultMode = new Mode(mode);
     AccessControlList acl = new AccessControlList();
     acl.mOwningUser = mOwningUser;
     acl.mOwningGroup = mOwningGroup;
     acl.mMode = mMode;
     if (mExtendedEntries == null) {
       acl.mExtendedEntries = null;
+      // minimal acl so we use defaultMode to filter user/group/other
+      acl.mMode = Mode.and(new Mode(mMode), defaultMode).toShort();
     } else {
+      // Rules for having extended entries, we need to modify user, mask and others'
+      // permission to be filtered by the defaultMode
       acl.mExtendedEntries = new ExtendedACLEntries(mExtendedEntries);
+
+      // mask is filtered by the group bits from the mode parameter
+      AclActions mask = acl.mExtendedEntries.getMask();
+      AclActions groupAction = new AclActions();
+      groupAction.updateByModeBits(defaultMode.getGroupBits());
+      mask.mask(groupAction);
+      // user is filtered by the user bits from the mode parameter
+      // other is filtered by the other bits from the mode parameter
+      Mode updateMode = new Mode(mMode);
+      updateMode.setOwnerBits(updateMode.getOwnerBits().and(defaultMode.getOwnerBits()));
+      updateMode.setOtherBits(updateMode.getOtherBits().and(defaultMode.getOtherBits()));
+      acl.mMode = updateMode.toShort();
     }
     return acl;
   }
 
   /**
    * Creates a child directory's access ACL and default ACL based on the default ACL.
+   * @param mode child's initial mode based on umask
```

Comment:
I'm not sure what "initial mode" means. Is this supposed to be the umask? If so, can we just call it the umask?

Reply:
changed it to umask